internal/schema/stacks/1.8/component_block.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package schema
+
+import (
+	"github.com/hashicorp/hcl-lang/lang"
+	"github.com/hashicorp/hcl-lang/schema"
+	"github.com/hashicorp/terraform-schema/internal/schema/refscope"
+	"github.com/hashicorp/terraform-schema/internal/schema/tokmod"
+	"github.com/zclconf/go-cty/cty"
+)
+
+func componentBlockSchema() *schema.BlockSchema {
+	return &schema.BlockSchema{
+		Description: lang.Markdown("Component represents the declaration of a single component within a particular Terraform Stack. Components are the most important object in a stack configuration, just as resources are the most important object in a Terraform module: each one refers to a Terraform module that describes the infrastructure that the component is 'made of'."),
+		Labels: []*schema.LabelSchema{
+			{
+				Name:                   "name",
+				SemanticTokenModifiers: lang.SemanticTokenModifiers{tokmod.Name},
+				Description:            lang.PlainText("Component Name"),
+			},
+		},
+		Body: &schema.BodySchema{
+			Extensions: &schema.BodyExtensions{
+				ForEach: true,
+			},
+			Attributes: map[string]*schema.AttributeSchema{
+				"source": {
+					Description:            lang.Markdown("The Terraform module location to load the Component from: a local directory (e.g. `./modules`), a git repository (e.g. `github.com/acme/infra/core`, `git::https://vcs.acme.com/acme/infra//core`), or a registry module (e.g. `acme-public/coreinfra/aws`, `app.terraform.io/acme/core-infra/aws`)"),
```

Comment:
I think private module sources are not supported yet – but I don't know if we need to reflect this.
Maybe this is our first difference between Terraform 1.8 vs 1.9 for example 😁 (Or maybe this is unrelated and will depend on the TFC Agent version.. 😵)